src/publish/fox_pub_pool.erl

```diff
@@ -160,3 +153,30 @@ terminate(_Reason, _State) ->
 -spec(code_change(term(), term(), term()) -> gs_code_change_reply()).
 code_change(_OldVersion, State, _Extra) ->
     {ok, State}.
+
+
+%% internal functions
+
+-spec handle_get_channel(queue:queue(pid()), pos_integer(), pid()) -> {pid(), queue:queue(pid())}.
+handle_get_channel(Channels, PoolSize, Connection) ->
+    NumChannels = queue:len(Channels),
+
+    {Channel1, Channels2} = case NumChannels < PoolSize of
+        true ->
+            {ok, Channel} = amqp_connection:open_channel(Connection),
+            {Channel, Channels};
+        false ->
+            {{value, Channel}, Channels1} = queue:out(Channels),
+            {Channel, Channels1}
+    end,
+
```

Comment:
It is not a big deal, however we can check if process alive only when we get it from queue. No need to check newly created channel.

Reply:
No need, it's not a big deal )